### PR TITLE
retain validation error

### PR DESCRIPTION
**Pull request by tpalmer**
So delegate view controllers can respond to different errors as needed.

### CODE REVIEW

File: SHEmailValidator/SHEmailValidationTextField.m [hunk 1]
```diff
@@ -158,7 +158,8 @@ - (void)commonInit
     self.messageDictionary = [NSMutableDictionary dictionary];
     self.defaultErrorMessage = @"Please enter a valid email address";
     self.messageForSuggestion = @"Did you mean";
-    
+    self.validationError = nil;
```
Comment (designatednerd) [inline SHEmailValidator/SHEmailValidationTextField.m]: In theory initializing this to `nil` shouldn't be necessary - if you don't give it any value to start with, it should already be nil. Have you run into an issue with that?

Comment (tpalmer) [inline SHEmailValidator/SHEmailValidationTextField.m]: No, this isn't strictly necessary - I can remove it if you'd prefer

**Comment by tpalmer** [comment]
@designatednerd I made that change if you want to take another look

**Comment by designatednerd** [comment]
LGTM. I'll hold off on a podspec bump until #11 is merged.

**Pull request merged**